src/perf/lib/Tcp.h

```diff
@@ -112,7 +112,10 @@ class TcpEngine {
 class TcpWorker {
     friend class TcpEngine;
     friend class TcpConnection;
+    CXPLAT_EXECUTION_CONTEXT ExecutionContext;
     bool Initialized{false};
+    bool IsExternal{false};
+    uint16_t PartitionIndex{0};
```

Comment:
Why is saving this necessary?

Reply:
WinKernel builds fail because since `CxPlatAddExecutionContext` is not defined in that environment, the function parameter `uint6_t AssignedCPU` is no longer referenced. This leads to a warning treated as an error. 

This code gives me an excuse to use `AssignedCPU` somewhere, which we may leverage later down the line.

Reply:
Copy how worker.c does this. It disabled calls to CxPlat for this on kernel mode. That's what you need to do. I don't think you'll need the variable saved at all.

Reply:
Ok, after some debugging with `valgrind`, looks like the reason we are running into SIGABRT (exit code 134) on Linux is because in the destructor of `Tcp.cpp`, we are attempting to free / wait on a non-existent thread. Currently, we use the `TcpWorker::Initialized` variable to determine whether or not to free/wait the thread. We should need the `TcpWorker::IsExternal` flag to avoid freeing a NULL pointer.